fingerprint2.js

```diff
@@ -316,181 +336,188 @@
         done(keys)
       })
     },
-    // kudos to http://www.lalit.org/lab/javascript-css-font-detect/
-    jsFontsKey: function (keys, done) {
+    jsFontsKey: function(keys, done, reject) {
       var that = this
       // doing js fonts detection in a pseudo-async fashion
       return setTimeout(function () {
-        // a font will be compared against all the three default fonts.
-        // and if it doesn't match all 3 then that font is not available.
-        var baseFonts = ['monospace', 'sans-serif', 'serif']
+        try {
+          that.jsFontsKeySynchronous(keys, done, that)
+        } catch(e) {
+          reject(e)
```

Comment:
Shouldn't there be a null check for `reject` here, or am I missing something?

Reply:
You are right :) By the way the "reject" parameter in get() function is not part of the public api, because errors within the get() function are not handed over to the reject().
This is on purpose to keep the current functionality and making my changes as small as possible.